tests/client/connstring.rs

```diff
@@ -0,0 +1,247 @@
+use mongodb::client::connstring;
+
+#[test]
+fn test_valid_uri() {
```

Comment:
including `test_` at the start of test functions isn't considered good style: you already have the attribute showing it's a test, and the runner uses the name to print it out, so you get output like

    test test_valid_uri

intead of

    test valid_uri